tools/terraform-bundle/package.go

```diff
@@ -306,26 +244,82 @@ not a normal Terraform configuration file. The file format looks like this:
   terraform {
     # Version of Terraform to include in the bundle. An exact version number
 	# is required.
-    version = "0.10.0"
+    version = "0.13.0"
   }
 
   # Define which provider plugins are to be included
   providers {
     # Include the newest "aws" provider version in the 1.0 series.
-    aws = ["~> 1.0"]
+    aws = {
+		versions = ["~> 1.0"]
+	}
 
     # Include both the newest 1.0 and 2.0 versions of the "google" provider.
     # Each item in these lists allows a distinct version to be added. If the
 	# two expressions match different versions then _both_ are included in
 	# the bundle archive.
-	google = ["~> 1.0", "~> 2.0"]
-	
-	#Include a custom plugin to the bundle. Will search for the plugin in the 
-	#plugins directory, and package it with the bundle archive. Plugin must have
-	#a name of the form: terraform-provider-*-v*, and must be built with the operating
-	#system and architecture that terraform enterprise is running, e.g. linux and amd64
-	customplugin = ["0.1"]
+	google = {
+		versions = ["~> 1.0", "~> 2.0"]
+	}
+
+	# Include a custom plugin to the bundle. Will search for the plugin in the 
+	# plugins directory, and package it with the bundle archive. Plugin must 
+	# have a name of the form: terraform-provider-*-v*, and must be built with
```

Comment:
```suggestion
	# have a name of the form "terraform-provider-<NAME>_v<VERSION>", and must be built with
```

I got caught out by this underscore/dash when testing locally. It led to a locally installed provider being silently ignored (the log level was `WARN`, which wasn't displayed by default).

Reply:
In local testing, I found that the provider was only correctly detected if it was structured like this:

`./plugins/example.com/myorg/customplugin/0.1/darwin_amd64/terraform-provider-customplugin_v0.1_linux_amd64.zip` (an archive containing the binary)

or:

`plugins/example.com/myorg/customplugin/0.1/darwin_amd64/terraform-provider-customplugin_v0.1` (the binary)

This seems different from this comment, and slightly different from the README.

Reply:
It turned out that everything about locally installed providers was broken, since fixed. I will update this comment though. I'm going to start using `terraform-provider-<NAME>`, even though there are other options that terraform can understand, since putting the version in the binary filename doesn't _add_ any value anymore.